RESTserver/src/main/java/com/softserve/if072/restservice/service/ForecastService.java

```diff
@@ -0,0 +1,316 @@
+package com.softserve.if072.restservice.service;
+
+import com.softserve.if072.common.model.Action;
+import com.softserve.if072.common.model.ProductStatistics;
+import com.softserve.if072.common.model.Storage;
+import com.softserve.if072.common.model.dto.AnalyticsProductDTO;
+import com.softserve.if072.common.model.dto.HistoryDTO;
+import com.softserve.if072.restservice.dao.mybatisdao.HistoryDAO;
+import com.softserve.if072.restservice.dao.mybatisdao.ProductDAO;
+import com.softserve.if072.restservice.dao.mybatisdao.StorageDAO;
+import com.softserve.if072.restservice.exception.NotEnoughDataException;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.lang.Math.round;
+
+/**
+ * The ForecastService class is used to calculate main descriptive statistics measures for using speed of product and
+ * forecast the end date of the product
+ *
+ * @author Igor Kryviuk
+ */
+@Service
+public class ForecastService {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final HistoryDAO historyDAO;
+    private final ProductDAO productDAO;
+    private final StorageDAO storageDAO;
+    private final ProductService productService;
+    private static final long MILLISECONDS_PER_DAY = 24 * 60 * 60 * 1000;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Cast one of the operands of this multiplication operation to a "long". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2184)